lib/engine/game/g_1848/step/buy_train.rb

```diff
@@ -17,14 +31,23 @@ def can_entity_buy_train?(entity)
           def buyable_trains(entity)
             # Cannot buy 2E if one is already owned
             owns_2e = entity.trains.any? { |t| t.name == '2E' }
-            return super unless owns_2e
-
-            super.reject { |t| t.name == '2E' }
+            trains_to_buy = super
+            trains_to_buy = trains_to_buy.select(&:from_depot?) unless @game.phase.status.include?('can_buy_trains')
+            trains_to_buy = trains_to_buy.reject { |t| t.name == '2E' } if owns_2e
+            trains_to_buy
           end
 
           def room?(entity)
             entity.trains.count { |t| t.name != '2E' } < @game.train_limit(entity)
           end
+
+          def spend_minmax(entity, train)
+            train_corp_owner = train.owner.owner
+            min = train_corp_owner == entity.owner ? 1 : train.price
+            max = train_corp_owner == entity.owner ? buying_power(entity) : train.price
```

Comment:
Easier to read if `return [1, buying_power(entity)] if train.owner.owner == entity.owner` and then `[train.price, train.price]`